### DISCUSSION TITLE
HX Stomp snapshot support

### DISCUSSION THREAD

**Discussion by potul** [alf45tar/PedalinoMini#385] [Ideas]
Hi, I've forked the project and I'm currently adapting it to my needs. One of the modifications I've done is to add a new type of action, named "Control Change Snap". The idea is to have an action that behaves like a Program Change (only one can be active at a time for a given channel and CC and when you stomp on it, the others go inactive), but using CC messages instead of PC messages. This is needed to use some "scenes" in certain devices like the HX stomp that uses "Snapshots"

I'm still testing the results but so far it's working fine.

If you are interested in the concept and want to add it let me know and I can provide more details.

**Comment by alf45tar** [comment]
Yes, I am. I will check you code but I suppose you used the same code for Program Change.
I didn't included it before because snapshots can be implemented using Sequences but it is nice to have in an easy way.
If you want to submit a PR I will include it into the public version and I will move to sponsors version too.

Thanks
alf45tar

**Reply by potul** [reply]
Ok, I will do some more testing tonight and it goes well I will submit a PR
And yes, I basically did a mashup between the Program change and the Control Change code.
I did not think on using Sequences for this, honestly. Maybe it's a cleaner option.

**Comment by potul** [comment]
Ok, I submitted a PR for this. Let me know if you need anything.

**Comment by potul** [comment]
After playing a little bit with it, I'm not sure the PR is worth it. Using sequences you can get the same with even more flexibility. The nice thing of this addition is you can deal with snapshots without sequences, but sequences are more flexible.
So up to you if you want to include it or not.

**Comment by alf45tar** [comment]
I already ported Control Code Snap into latest version 3.0.1 using your contribution with few changes. I will not merge your PR but it remains public for whom do not want to wait next public release.

Thanks

**Comment by alf45tar** [comment]
@potul if you want to share your build I will be happy to add a photo to PedaliniMini home page.

**Reply by potul** [reply]
I'll take some pictures tonight and send them.

**Reply by potul** [reply]
I just opened a new thread with some pictures.
BTW, how do I access the sponsors repo? Do I need to wait for some kind of approval after subscribing?

**Reply by alf45tar** [reply]
Yes, I have to add you as collaborators of the private repository https://github.com/pedalino-sponsors/PedalinoMini.
Did you received the email?

**Reply by potul** [reply]
yes, thanks. Already looking at it. :)
